### PR TITLE
feat: oidc4ci js api

### PR DESCRIPTION
**Pull request by Moopli**
Signed-off-by: Filip Burlacu <filip.burlacu@securekey.com>

### CODE REVIEW

**Comment by Moopli** [comment]
note, still finishing up tests

File: cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js
Comment (rolsonquadras) [inline cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js:216]: We should give the control to consumer to poll as deferred flow could take days as well.

**Review by rolsonquadras — APPROVED** [review]

File: cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js
Comment (birtony) [inline cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js:143]: Since you don't seem to need the url itself, why not use the `URLSearchParams` instead?

Another thing is that it returns `null` if the parameter does not exist, so you'd need to check for that instead:

File: cmd/wallet-js-sdk/src/index.js
Comment (birtony) [inline cmd/wallet-js-sdk/src/index.js:31]: I guess we need to update the header on top of this file

File: cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js
Comment (birtony) [inline cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js:282]: Should it be `base64` or `base64url` encoded?

File: cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js
Comment (birtony) [inline cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js:188]: This could be a `const`

File: cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js
Comment (birtony) [inline cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js:210]: This could be a `const`

**Review by birtony — COMMENTED** [review]
Another thing that may be worth having a discussion is the use of `camelCase` and `snake_case` in function names and local variables - I think overall we used to stick to camel case in this project

File: cmd/wallet-js-sdk/src/index.js
Comment (Moopli) [reply cmd/wallet-js-sdk/src/index.js:31]: Don't think so? It already has the avast copyright notice.

File: cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js
Comment (Moopli) [reply cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js]: https://github.com/trustbloc/agent-sdk/pull/452/files#diff-b6fc870019a039dfefb1922ebe309e51614712b835bd26d6b2a6d79c09fb8aa1R45-R46
good?

File: cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js
Comment (Moopli) [reply cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js]: removed, this api doesn't use this client config parameter.

File: cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js
Comment (Moopli) [reply cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js]: done https://github.com/trustbloc/agent-sdk/pull/452/files#diff-b6fc870019a039dfefb1922ebe309e51614712b835bd26d6b2a6d79c09fb8aa1R130

File: cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js
Comment (Moopli) [reply cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js:245]: either is fine, this isn't for a URL

**Review by birtony — APPROVED** [review]
Looks good

**Comment by birtony** [comment]
Let's wait for @sudeshrshetty to review before merging

File: cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js
Comment (sudeshrshetty) [inline cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js:35]: just curious,

Should default signer throw error instead?

File: cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js
Comment (birtony) [inline cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js:150]: Maybe just using `new URLSearchParams()` would be cleaner?

File: cmd/wallet-js-sdk/test/specs/oidc/issuance/oidc4ci.spec.js
Comment (birtony) [inline cmd/wallet-js-sdk/test/specs/oidc/issuance/oidc4ci.spec.js:94]: Should this be declared at the top of the file before tests?

File: cmd/wallet-js-sdk/test/specs/oidc/issuance/oidc4ci.spec.js
Comment (birtony) [inline cmd/wallet-js-sdk/test/specs/oidc/issuance/oidc4ci.spec.js:151]: Since this could be reused?

File: cmd/wallet-js-sdk/test/specs/oidc/issuance/oidc4ci.spec.js
Comment (Moopli) [reply cmd/wallet-js-sdk/test/specs/oidc/issuance/oidc4ci.spec.js:3]: ???

**Review by sudeshrshetty — APPROVED** [review]

File: cmd/wallet-js-sdk/src/index.js
Comment (Moopli) [reply cmd/wallet-js-sdk/src/index.js]: Good point, I refactored these functions into an OpenID4CI client object.

File: cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js
Comment (Moopli) [reply cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js]: Done.

**Review by birtony — APPROVED** [review]

**Pull request merged**
